app/code/core/Mage/Sales/Model/Order/Payment.php

```diff
@@ -806,7 +806,7 @@ public function registerRefundNotification($amount)
             $amount = $amountRefundLeft;
         }
 
-        if ($amount != $baseGrandTotal) {
+        if (bccomp($amount, $baseGrandTotal, 2) != 0) {
```

Comment:
I think using `Mage::helper('core')->getExactDivision($amount, $baseGrandTotal) === 0` will work here, as it also has a precision of 2 decimal digits.

My question is, the values in the DB have a precision of 4 decimal digits, so is it safe to assume that no one cares about the last two?

Reply:
Moreover some currencies use [more than 2 digits](https://docs.adyen.com/development-resources/currency-codes/), for example _Jordanian Dinar / JOD_.
Not sure if Magento round every currencies to 2 digits or not.

`getExactDivision` give the same results:
```php
echo "getExactDivision\n";
var_dump(Mage::helper('core')->getExactDivision(10.01, 10.01 + 0.01) === 0);
var_dump(Mage::helper('core')->getExactDivision(10.01, 10.01 + 0.001) === 0);
var_dump(Mage::helper('core')->getExactDivision(10.01, 10.01 + 0.0001) === 0);
var_dump(Mage::helper('core')->getExactDivision(10.01, 10.01 + 0.00001) === 0);

echo "bccomp\n";
var_dump(bccomp(10.01, 10.01 + 0.01, 2) == 0);
var_dump(bccomp(10.01, 10.01 + 0.001, 2) == 0);
var_dump(bccomp(10.01, 10.01 + 0.0001, 2) == 0);
var_dump(bccomp(10.01, 10.01 + 0.00001, 2) == 0);
```

Reply:
@kyrena Beware on strict type comparison: https://onlinephp.io/c/4af76
```php
$var = [0 === 0.0, 0 == 0.0];
var_dump($var);
```
Result:
```
array(2) {
  [0]=>
  bool(false)
  [1]=>
  bool(true)
}
```